src/components/PasswordSetterBox.js

```diff
@@ -107,32 +116,59 @@ class PasswordSetterBox extends Nimiq.Observable {
         this.$el.classList.remove('repeat');
     }
 
-    /**
-     * @returns {Promise<void>}
-     */
     async onPasswordIneligible() {
-        // We have to shake both possible too-weak notices
         const $hintTooShort = /** @type {HTMLElement} */ (this.$el.querySelector('.password-strength.strength-short'));
-        const $hintWeak = /** @type {HTMLElement} */ (this.$el.querySelector('.password-strength.strength-weak'));
-
-        await Promise.all([
-            AnimationUtils.animate('shake', $hintTooShort),
-            AnimationUtils.animate('shake', $hintWeak),
-        ]);
+        await AnimationUtils.animate('shake', $hintTooShort);
     }
 
     /**
      * @param {boolean} isValid
      */
     _onInputChangeValidity(isValid) {
-        if (this._password && this._passwordInput.text === this._password) {
-            this.fire(PasswordSetterBox.Events.SUBMIT, this._password);
-            return;
+        if (this._repeatPasswordTimout) {
+            window.clearTimeout(this._repeatPasswordTimout);
+            this._repeatPasswordTimout = null;
         }
 
+        if (this._password) {
+            if (this._passwordInput.text === this._password) {
+                this._repeatPasswordTimout = window.setTimeout(
+                    () => {
+                        this.fire(PasswordSetterBox.Events.SUBMIT, this._password);
+                        this._passwordInput.reset();
+                    },
+                    400,
+                );
+                return;
+            }
+            if (this._passwordInput.text.length > 0) {
+                if (this._passwordInput.text.length < this._password.length) {
+                    this._repeatPasswordTimout = window.setTimeout(
+                        async () => {
+                            this.$el.classList.remove('repeat-long');
+                            this.$el.classList.add('repeat-short');
+                            await AnimationUtils.animate('shake', this._passwordInput.$el);
```

Comment:
Please remove this shake here. It does not make any sense, as the password entry is not finished yet.